proteus/Domain.py

```diff
@@ -59,6 +63,144 @@ def getBoundingBox(self):
             self.L += [xMax-xMin]
 
 
+    def writeGeo(self, fileprefix, group_names=False):
```

Comment:
It should work for all domain classes as the `.geo` files use the same format for 1D, 2D, or 3D domains. I just realised that `D_Base` does not have `segments`, `facets`, etc. defined in `__init__` (as I usually use SpatialTools to define a Domain), would it be okay to populate those with empty lists `[]` as a default? This way, the `writeGeo` will write what has been defined and ignore empty lists.